### ISSUE TITLE
Allow `y` value for horizontal line, rather than index

### ISSUE THREAD

**Issue by nandorojo** [coinjar/react-native-wagmi-charts#26]
Currently, the `at` prop for `LineChart.HorizontalLine` uses an x value. However, you should be able to set a `y` value here instead.

Take the case of Robinhood. The horizontal line doesn't correspond to any data point on the chart. Instead, it represents the previous day's trading price. This is important for day-to-day growth comparisons.

<img src="https://user-images.githubusercontent.com/13172299/139606913-9abd6eaf-7882-414e-8d37-beab70be6a32.jpeg" width="250" />

What if we amend the `HorizontalLine`'s `at` prop to take _either_ an `x` or a `y`?

```ts
type At = { x: number; y?: never } | { x?: never; y: number }
```

**Comment by jxom** [comment]
Yeah, I think this is a good idea. The `at` prop currently is the **index** of the data (not the timestamp (x) or value (y) values). I guess you would have to approach this from two angles:

1. A user may still want to use the functionality the current `at` prop provides – i.e. specifying an **index**,
2. A user may want to add the horizontal line at a particular `y` value.

To implement both approaches, I think specifying an object to `at` like you mentioned may be a good idea. I'm not too fond of creating a breaking change this early in the library, so maybe the interface could look like:

```js
type At = number | { index: number; value?: never } | { index?: never, value: number }
```

- I renamed `x` to `index` to be compatible with the current implementation,
- I renamed `y` to `value` so it mirrors to the schema of the input `data`,
